pontoon/base/static/js/fluent_interface.js

```diff
@@ -198,25 +249,69 @@ var Pontoon = (function (my) {
       },
 
 
+      /*
+       * Is ast of a supported message?
+       *
+       * Message is supported if all value elements
+       * and all attribute elements are of type:
+       * - TextElement
+       * - ExternalArgument
+       * - MessageReference
+       * - SelectExpression?
+       */
+      isSupportedMessage: function (ast) {
+        function elementsSupported(elements) {
+          return elements.every(function(item) {
+            return [
+              'TextElement',
+              'ExternalArgument',
+              'MessageReference',
+              'SelectExpression'
+            ].indexOf(item.type) >= 0;
```

Comment:
Shouldn't that be `=== 0`? Otherwise if there's a `FooTextElement` in there, it will considered supported, but it might have unexpected side effects.

Reply:
https://github.com/mozilla/pontoon/pull/814/commits/46c9f2c601fd96ff15c8699665a60dc7e4b5d44e :)

Reply:
Ahahaha, sorry! 😄